R/Utils.R

```diff
@@ -116,4 +175,16 @@ RenameUsingCD <- function(inputGenes) {
   toMerge <- arrange(toMerge, SortOrder)
 
   return(toMerge$Name)
-}
\ No newline at end of file
+}
+
+#' @title Update Seurat Object Gene Model From Monkey LOCs to Human Gene IDs
+#'
+#' @description Uses UpdateGeneModel to change LOC genes to more common human gene IDs in the RNA assay
+#' @param seuratObj The Seurat Object to be updated
+#' @export
+
+seuratFeaturesMonkeyToHuman = function(seuratObj){
```

Comment:
can you please add an assayName = 'RNA' argument, and use that variable in the function? Having a customizable argument with a default (so it can generally be ignored) is often good practice as opposed to always assuming values in the function

Reply:
and what about scale.data? perhaps you can iterate the slots programmatically instead of hard-coding?

Reply:
this is a bit picky, but we should rename the function. It's not really projecting to human per se. It's correcting adding gene symbols to genes in NCBI that we believe simply were not annotated with a gene symbol when they should be. Perhaps: UpdateMacaqueNcbiGeneSymbols? 

Note: good use above with the "." for internal methods. Please capitalize the exported ones too.

Reply:
Regarding programmatically figuring out slot names: I would perhaps just make a list with the strings counts, data, and scale.data. Iterate those, check whether that slot exists, and if it does then update it.

Reply:
Note: you might also want to think about the potential for other assays, like integrated data. Perhaps just iterate every assay and change that feature whenever present? Even if ADTs never have them, it's a minor amount of work for the peace of mind knowing you get them all. This would negate my earlier command about making an assayName argument (though not the rationale for using arguments like this when appropriate)

Reply:
Thanks for the feedback! I've updated the function name to UpdateMacaqueMmul10NcbiGeneSymbols to enforce the specificity of the function to just Mmul_10. 

Additionally I've updated the function to scour the assays and slots in the seurat object, specifically to update the counts, data, and scale.data slots. These can either be S4 (default by seurat) or a matrix (for example, seuratObj@assay$ADT@data is a matrix after dsb normalization in our ADT experiments). 

`attr(x = seuratObj@assays[[assay]], which = slot)` seems to work well enough for programmatically accessing the slots. 

In order to keep `.UpdateGeneModel()` simple and easy to add more genes, I made a lot of assumptions on each assay & slot combination. These will likely break in the future as seurat updates, so I've added a verbose tag as well. 

I also added the gene translations that were present in Phenotyping.R. Pulling this function into the pipeline will break the phenotyping plots, so they'll need to be updated. It's under 10 or so names, so not a big deal. I can do this later when we decide exactly how we want to use this.